lib/util.js

```diff
@@ -970,17 +970,10 @@ exports.inherits = function(ctor, superCtor) {
   Object.setPrototypeOf(ctor.prototype, superCtor.prototype);
 };
 
-exports._extend = function(target, source) {
-  // Don't do anything if source isn't an object
-  if (source === null || typeof source !== 'object') return target;
-
-  var keys = Object.keys(source);
-  var i = keys.length;
-  while (i--) {
-    target[keys[i]] = source[keys[i]];
-  }
-  return target;
-};
+exports._extend =
+  internalUtil.deprecate(
+    internalUtil._extend,
+    'util._extend is deprecated. Use Object.assign() instead');
```

Comment:
I'm not sure we should be explicitly advocating for `Object.assign()` since it's still quite slower (IIRC) than the loop-based solution used in `internalUtil._extend()`.